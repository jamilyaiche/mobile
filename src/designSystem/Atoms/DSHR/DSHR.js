import React from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import Style from '../../../assets/styles/Style'

const DSHR = (props) => {
  let size = Style.hrMedium
  if (props.size === 'small') {
    size = Style.hrSmall
  } else if (props.size === 'large') {
    size = Style.hrLarge
  }

  let theme = Style.hrDefault
  if (props.theme === 'primary') {
    theme = Style.hrPrimary
  }
  return (

    <View
      style={[
        size,
        theme,
        {
          width: props.width === 'full' ? '100%' : props.width
        }
      ]}
    >

    </View>
  )
}

DSHR.defaultProps = {
  size: 'medium',
  theme: 'default',
  width: 'full'
}

DSHR.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  theme: PropTypes.oneOf(['default', 'primary']),
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}

export default DSHR
