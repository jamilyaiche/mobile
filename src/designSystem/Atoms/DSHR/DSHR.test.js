import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { Text, TouchableOpacity, TextInput as RNTextInput, Pressable } from 'react-native'
import DSHR from './DSHR'
jest.mock('rn-fetch-blob', () => {
    return {
      fs: {
        dirs: {
          DocumentDir: ''
        },
        writeFile: () => Promise.resolve()
      }
    }
  })
  jest.mock('react-native-x2-carousel', () => {
    return {
      fs: {
        dirs: {
          DocumentDir: ''
        },
        writeFile: () => Promise.resolve()
      }
    }
  })
describe('DSHR tests', () => {
  it('test default DSHR snapshot', () => {
    const wrapper = renderer.create(<DSHR/>)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
  it('test DSHR props snapshot', () => {
    const wrapper = renderer.create(<DSHR  
    size= 'medium'
    theme = 'primary'
    width= '50%'
    />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
  
 
})
