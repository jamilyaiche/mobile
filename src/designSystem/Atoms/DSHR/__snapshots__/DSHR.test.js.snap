// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSHR tests test DSHR props snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "height": 2,
      },
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "width": "50%",
      },
    ]
  }
/>
`;

exports[`DSHR tests test default DSHR snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "height": 2,
      },
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "width": "100%",
      },
    ]
  }
/>
`;
